django_ilmoitin/admin.py

```diff
@@ -0,0 +1,30 @@
+from django.contrib.admin import site as admin_site
+from parler.admin import TranslatableAdmin
+from parler.forms import TranslatableModelForm
+
+from .models import NotificationTemplate
+
+
+class NotificationTemplateForm(TranslatableModelForm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Do not allow the admin to choose any of the template types that already
+        # exist.
+        qs = NotificationTemplate.objects.values_list("_type", flat=True)
+        try:
+            if self.instance and self.instance.type:
+                qs = qs.exclude(id=self.instance.id)
+        except ValueError:
+            pass
+        existing_types = set(enum for enum in qs)
+        choices = [
+            x for x in self.fields["_type"].choices if x[0] not in existing_types
```

Comment:
since you are just doing `set(enum for enum in qs)`, you could delete the `existing_types` var completely and just do:
```python
choices = [
    x for x in self.fields["_type"].choices if x[0] not in set(qs)
]
```